syntax = "proto3";

package llm;

import "google/protobuf/empty.proto";

service LLMService {
  rpc ProcessTextStream(stream LLMStreamRequest) returns (google.protobuf.Empty);
}

message LLMStreamRequest {
  oneof request_data {
    SessionConfig config = 1;
    string text_chunk = 2;
  }
}

message SessionConfig {
  string session_id = 1;          // LLM 서비스 내부에서 사용할 수 있는 세션 ID (선택적)
  string frontend_session_id = 2; // ★ 새로 추가: 프론트엔드 웹소켓 세션 ID
  // string model_name = 3;
  // string user_prompt_template = 4;
}
